SofaKernel/modules/SofaCore/src/sofa/core/Multi2Mapping.inl

```diff
@@ -224,18 +212,15 @@ void Multi2Mapping<In1,In2,Out>::computeAccFromMapping(const MechanicalParams* m
 }
 
 template < class In1, class In2, class Out >
-void Multi2Mapping<In1,In2,Out>::init()
+void Multi2Mapping<In1, In2, Out>::init()
 {
-    maskFrom1.resize( this->fromModels1.size() );
-    for( unsigned i=0 ; i<this->fromModels1.size() ; ++i )
-        if( core::behavior::BaseMechanicalState* stateFrom = this->fromModels1[i]->toBaseMechanicalState() ) maskFrom1[i] = &stateFrom->forceMask;
-    maskFrom2.resize( this->fromModels2.size() );
-    for( unsigned i=0 ; i<this->fromModels2.size() ; ++i )
-        if( core::behavior::BaseMechanicalState* stateFrom = this->fromModels2[i]->toBaseMechanicalState() ) maskFrom2[i] = &stateFrom->forceMask;
-    maskTo.resize( this->toModels.size() );
-    for( unsigned i=0 ; i<this->toModels.size() ; ++i )
-        if (core::behavior::BaseMechanicalState* stateTo = this->toModels[i]->toBaseMechanicalState()) maskTo[i] = &stateTo->forceMask;
-        else this->setNonMechanical();
+    for (unsigned i = 0; i < this->toModels.size(); ++i)
+    {
+        if (!this->toModels[i]->toBaseMechanicalState())
+        {
+            this->setNonMechanical();
```

Comment:
break?
and what happens if one of the `toModels` is a `MechanicalState` but not the other?

Not related to this PR.

Reply:
I was thinking the same while doing this...